conjure-codegen/src/lib.rs

```diff
@@ -480,7 +568,15 @@ impl ModuleTrie {
         Ok(())
     }
 
-    fn create_root_module(&self) -> TokenStream {
+    fn create_root_module(&self, lib_root: bool) -> TokenStream {
+        let attrs = if lib_root {
+            quote! {
+                #![allow(warnings)]
```

Comment:
Why do we want to allow warnings if `lib_root`?

Reply:
Since we're autogenerating code it can end up triggering things like clippy, and it's easier to just turn that off rather than trying to make everything way smarter on the generation side.

Reply:
`create_root_module` shouldn't need to know about `lib_root`. How about defining a `fn allow_warnings() -> TokenStream` that we can `extend` with the `TokenStream` returned by `create_root_module`? This moves `lib_root` to the callers and simplifies `create_root_module`.

Reply:
It seems to make sense to me that the thing that's responsible for knowing how to create the root module for a package would want to know what kind of root it is.

Reply:
Sure, no strong preference. I prefer separate methods `create_root_lib` and `create_root_module` if we're making the distinction at this level.